### DISCUSSION TITLE
v3.2.2 (Beta)

### DISCUSSION THREAD

**Discussion by osy** [utmapp/UTM#3997] [Releases]
## Installation
**iOS**: Visit [https://getutm.app/install/][1] for the most up to date installation instructions.
**macOS**: Download "UTM.dmg" below and copy the contents to your `/Applications` directory.

## Highlights
* **QEMU 7.0.0**: The backend has been updated to the latest release. Check out https://www.qemu.org/2022/04/19/qemu-7-0-0/ for more details. Additionally, other backend components such as SPICE GTK, libusbredir, and more have also been updated.
* **Reclaiming space**: the host will attempt to TRIM any page of all zeros when committed to the QCOW2 disk image. Additionally, a new button in disk settings will allow you to re-convert a disk image to reclaim space (of all-zero pages) as well as compress an existing disk image (macOS 12+ only). Note compression is only done statically which means when the guest modifies a sector, it will be stored uncompressed just like before.
* **Disposable mode**: right click/long press on a QEMU VM and select "Run without saving changes" to start VM in "disposable mode." In this mode, changes will not be saved to disk and will be discarded once the VM is stopped. (thanks @ktprograms)

## Notes
* (iOS) **UTM v3.2.x will be the last update to support iOS 11-13.** If you are still running those versions, please make sure you report any issues now so they can be fixed by the final release of v3.2.x. After that, iOS 14+ will be required to run future versions of UTM.
* In v3.2.0, if you added `highmem=on` to your QEMU Machine Properties as a workaround, please remove it after updating. If you did not use v3.2.0, you can ignore this.

## Changes (v3.2.2)
* Add "disposable mode" (see highlights for more information) (thanks @ktprograms)
* Do not enable UEFI for unsupported architectures (#3976)
* Fix broken play button in sidebar (#3980)
* Fix "Error -1 while loading VM state" on resume caused by usbredir (#3979)
* Various UI tweaks and typo fixes (#3984)
* Fixed pausing without saving state
* Fixed showing warning prompt when VM is paused but not saved
* (iOS) Detect if UTM is installed without AltStore and AltJIT will not attempt to be used.
* (iOS) Reduced build size by 10% (thanks @Halo-Michael) (#3934)
* (macOS) Add new preference option to revert Caps Lock behaviour to pre 3.2.0 (#3974)

## Changes (v3.2.1)
* Fixed boot of RISC-V virt platform (thanks @wegank)
* Remove `highmem=off` default property which is no longer needed on macOS 12.4+ and iOS 15.5+ (it will still be used on lower versions) (#3941, #3946)
* Caps lock state will now be synchronized when mouse is captured (note that if mouse is captured before the guest OS has booted up, the state will be out of sync) (#3415)
* Re-introduce option to disable auto-resolution when SPICE tools is installed (#3938)
* When discarding unsaved settings, do not delete any disk image not created by UTM (#3587)
* Fix downloading of UTM zip files when the URL does not end in ".zip" (#3953)
* (macOS) When importing a disk image, do not silently overwrite an existing image of the same name (#3866)
* (macOS) Added option to reclaim unused space from disk image and to compress a disk image (macOS 12+ only)
* (macOS) On the VM list, the start button will now change to a spinner when the VM is starting and will disappear when the VM has started. Also added the ability to start/show a VM by double-clicking on the list item (#3885)

## Changes (v3.2.0)
* Checkbox for "Import VHDX Image" in the wizard now works correctly again (#3760)
* Fixed various crashes in SPICE webdavd share has been resolved thanks to updated backend (#3633)
* Removed "cache=writethrough" argument from block devices since it was added to workaround an issue with VHDX images which we now convert to QCOW2
* Added "discard=unmap,detect-zeroes=unmap" argument to block devices to reclaim unused space (#3460)
* When "do not save screenshot to disk" is enabled, any existing screenshot will be deleted the next time the VM is started (#3936)
* Fixed an issue in QEMU additional arguments where it sometimes would not appear after setting it (#3895)
* Fixed RNG device not appearing in console mode even when enabled in the settings (#3790)
* Added support for up to 5 additional mouse buttons (you may need to re-map them in the guest OS as they are automatically set to middle/up/down/side/extra) (#3906)
* (iOS) Fixed an issue in the legacy view where "invalid identifier" error may show up on a newly created VM (#3872)
* (macOS) Fixed crash when changing display settings while the VM window is open (but stopped) (#3925)
* (macOS) Fixed crash when trying to select "Share..." in context menu (#3915)
* (macOS) Prevent allowing rapid clicking of pause/resume which may lead to VM corruption (#3629)
* (macOS) Show progress indicator while VM is saving to prevent user from terminating the app early and cause corruption (#3752)
* (macOS) Block unsupported Apple USB devices from being captured (#3759)

## Issues
Please check the full list on [Github][2] and help report any bug you find that is not listed.

### Known Issues

* iOS 14 and below: There is a known issue crashing while running under taurine but not unc0ver. (#3628)

[1]: https://getutm.app/install/
[2]: https://github.com/utmapp/UTM/issues

<hr /><em>This discussion was created from the release <a href='https://github.com/utmapp/UTM/releases/tag/v3.2.2'>v3.2.2 (Beta)</a>.</em>

**Comment by JacksonChen666** [comment]
i think it would be better if the user was able to discard the vm session on demand when stopping it instead of choosing so at the start, so that they don't have to make a decision for the future as the future is mostly uncertain

**Reply by osy** [reply]
So it doesn’t work that way because as you’re modifying the VM, changed are being written. Otherwise what if the power goes out or your computer crashes? You lose everything?

**Reply by JacksonChen666** [reply]
@osy i'd suggest something that works *like* lvm snapshots (but probably not *actually* lvm snapshots on the host system)

**Reply by ktprograms** [reply]
This might be achievable (slightly hacky) by using qcow2 snapshots:
- Create a snapshot before starting the VM
- After the VM has stopped, show dialog to either:
  * Remove pre-start snapshot, causing the earlier state to be lost - normal non-disposable mode
  * Revert to pre-start snapshot - disposable mode

The only thing to watch out for is deleting the new state after reverting, otherwise a lot of additional space will be used over time.

**Reply by ssbarnea** [reply]
Can someone tell me where is the missing DMG to download? The page at https://github.com/utmapp/UTM/releases/tag/v3.2.2 mentions "Download "UTM.dmg" below " but the is no link to it.

The latest DMG version I seen is 3.1.5 and the app itself has not check for updates option.

**Reply by JacksonChen666** [reply]
@ssbarnea the assets section of the release so where you'll find all the downloads
you may have to expand it

(you're also replying to an unrelated comment thread)

**Comment by ssbarnea** [comment]
Can we do something to improve the download experience?

* newer builds are not advertised in any way on official installation website
* the page from https://github.com/utmapp/UTM/releases mentions in the second bullet "Download "UTM.dmg" below but i did scroll a lot and was not able to spot where was the download link. One hour later I was able to stop the fact that there is a collapsed assets section at the bottom of the release box,... very well hidden.
* the client itself is unable to check for newer updates
* the page at https://mac.getutm.app/ does not mention current version or any link to beta downloads, it only has download latest permalink, so if you want a newer version, you are expected to download it again and check which version it is by running it. i know that updating the website each time may seem like extra burden, but how about putting a link to github releases from the webside, at least would inform the user where he can find more info and versions.

**Reply by osy** [reply]
Beta are unstable so it’s not advertised specifically so people who are unaware won’t download it. The collapsed assets seemed to be a recent GitHub change. I’ll contact them about it.

**Reply by osy** [reply]
https://github.com/github/feedback/discussions/16379
